contracts/ParsiqPancakeConverter.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+
+// IQ Protocol. Risk-free collateral-less utility renting
+// https://iq.space/docs/iq-yellow-paper.pdf
+// (C) Blockvis & PARSIQ
+// 🖖 Stake strong!
+
+pragma solidity 0.8.4;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./interfaces/IConverter.sol";
+import "./libs/Errors.sol";
+import "./libs/uniswap-v2/IUniswapV2Router02.sol";
+import "./libs/uniswap-v2/IUniswapV2Pair.sol";
+import "./libs/uniswap-v2/IUniswapV2Factory.sol";
+
+/**
+ * Pancakeswap converter for estimating token prices.
+ */
+contract ParsiqPancakeConverter is IConverter {
+    IUniswapV2Pair public swapPair;
+    IUniswapV2Router02 private _uniswapRouter;
+
+    /**
+     * @notice Constructor for `ParsiqPancakeConverter`
+     * @param uniswapRouter - UniswapV2 router implementation. On BSC that would be Pancakeswap.
+     * @param allowedTokenOne - The ERC20 token that's used for finding the swap pair
+     * @param allowedTargetTwo - The ERC20 token that's used for finding the swap pair
+     * @dev The token pair must be pre-deployed and registered on the router!
+     * @dev The contract will find the existing token pair.
+     */
+    constructor(
+        IUniswapV2Router02 uniswapRouter,
+        IERC20 allowedTokenOne, // <- STABLECOIN (BUSD)
+        IERC20 allowedTargetTwo // <- PRQ
+    ) {
+        _uniswapRouter = uniswapRouter;
+        IUniswapV2Factory uniswapFactory = IUniswapV2Factory(_uniswapRouter.factory());
+        swapPair = IUniswapV2Pair(uniswapFactory.getPair(address(allowedTokenOne), address(allowedTargetTwo)));
+    }
+
+    /**
+     * @notice Perform estimation of how many `target` tokens are necessary to cover required amount of `source` tokens.
+     * @param source - the source token address (e.g. stablecoin)
+     * @param target - the target token address (e.g. volitale coin)
+     * @param amountInSourceTokens - the amount of source token price
+     * @dev Source and target token addresses must be the exact ones as specified in the constructor of this contract.
+     */
+    function estimateConvert(
+        IERC20 source,
+        uint256 amountInSourceTokens,
+        IERC20 target
+    ) external view override returns (uint256) {
+        address pairToken0 = swapPair.token0();
```

Comment:
Please add check for `source == target` and return `amountInSourceTokens` if it is true

Reply:
Alright, added it. Although this also means that now you can pass tokens to the converter that initially were not registered and no exception will be raised.